### PR TITLE
add .email top-level domain

### PR DESCRIPTION
**Pull request by bjfletcher**
ben@fletcher.email to get a green light

### CODE REVIEW

**Comment by mmockett** [comment]
Is there any benefit on whitelisting these instead of just accepting any set of alphabetical characters after the dot? Surely the number of domain extensions is very large, and will keep growing.

**Comment by bjfletcher** [comment]
@mmockett I think it brings a lot of benefit to inform the user as soon as possible if the email address is incorrect, so if you type "ben@gmail.comm" and then hit return, by accident, you would like to be informed at this point, rather than later on when your contact tells you that they've not received any email from you.

Error rate for email entry is very high, which is why most websites ask you to enter the email address twice, in two fields.

**Comment by mmockett** [comment]
True. What's the scope of the list though - should it contain all [these](http://www.iana.org/domains/root/db)?

**Comment by i-like-robots** [comment]
This regex could cause more issues than it solves... missing TLDs being one but also the internationalized domain names which isn't unlikely given our customer base.

AFAIK the best regex for validation email is just `/@{1}/`!

File: data/actions.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ function hideKeyboard () {
 
 function validateEmail (email) {
 	// return true if invalid
-	return email !== '' && !email.match(/^[-a-z0-9~!$%^&*_=+}{\'?]+(\.[-a-z0-9~!$%^&*_=+}{\'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|gov|info|int|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i)
+	return email !== '' && !email.match(/^[-a-z0-9~!$%^&*_=+}{\'?]+(\.[-a-z0-9~!$%^&*_=+}{\'?]+)*@([a-z0-9_][-a-z0-9_]*(\.[-a-z0-9_]+)*\.(aero|arpa|biz|com|coop|edu|email|gov|info|int|mil|museum|name|net|org|pro|travel|mobi|[a-z][a-z])|([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}))(:[0-9]{1,5})?$/i)
```
Comment (matthew-andrews) [inline data/actions.js:11]: i think we should probably just make this much more relaxed………

there are so many valid domains nowadays:-
https://iwantmyname.com/?domain=benfletcher

screen shot 2016-06-14 at 11 46 09

Comment (bjfletcher) [inline data/actions.js:11]: How'd we ensure we catch common errors like "ceo@ft.comm" if we're going to allow all that?

Comment (matthew-andrews) [inline data/actions.js:11]: i suggest that we don't and let users take responsibility for that :-)

Comment (matthew-andrews) [inline data/actions.js:11]: e.g. `.co` is a valid domain

Comment (i-like-robots) [inline data/actions.js:11]: How about we suggest common email domains, perhaps that would also help prevent mistakes?

**Comment by mmockett** [comment]
We've had another complaint about this (this time about a .international domain). Can we go with Matt's suggested `/@{1}/` approach for now and address it properly later if we see issues with bounces?

**Comment by bjfletcher** [comment]
I think I'll go with this regex:

`/[^@]{1,}@[^@]{1,}\.[^@]{1,}/`

which ensures there's only one @ and at least one dot after the @

**Comment by matthew-andrews** [comment]
:v:

**Pull request merged**